### PR TITLE
Add recombinant clade definition

### PR DESCRIPTION
**Pull request by joverlee521**
Adding "recombinant" to clade definition to fix the AssertionError raised by our automated model runs.  

Follow up to #67 

## Checklist

- [ ] Checks pass

### CODE REVIEW

File: config/config.yaml [hunk 1]
```diff
@@ -124,4 +124,8 @@ clade_definitions:
   - clade: "other"
     display_name: "other"
     defining_lineage: False
-    color: '#777777'
\ No newline at end of file
+    color: '#777777'
+  - clade: "recombinant"
+    display_name: "recombinant"
+    defining_lineage: False
+    color: '#515151'
```
[the remark this answers is not in the document]
Comment (jameshadfield) [reply config/config.yaml:131]: Re: colour, here's how the clades output looks with this change, which I think is appropriate given that recombinant is similar in spirit to "other". Prior to #67 clades without defined colours would have been greyscale, and recombinant was never assigned a colour, so this is consistent with that outcome. However I have bigger questions about how we treat the "recombinant" category...

image

**Review by jameshadfield — CHANGES_REQUESTED** [review]
The current PR works well in the context of `nextstrain_clades` however I think the existance of "recombinant" causes a mismatch between this and `pango_lineages` which I want to point out here. For the latter, we try to assign lineages to their respective nextstrain clade with "other" being the fallback. This means no lineages will be assigned to "recombinant". I can see a few different options here:

* Modify the upstream data categorization so that what's currently "recombinant" is grouped together with "other". This seems the most sensible to me but I am not an expert on the ingest side of things here. 
* Modify the lineage-colouring script so that lineages which are not part of a named nextstrain clade are assigned to "other" or "recombinant" by checking if the pango lineage starts with "X". 
* Do nothing -- perhaps this doesn't matter, especially if we use black as the colour assigned to "recombinant" and grey for "other" then the lineages look like this:

image

(I assume it's XBC* and XCH which are grouped as "recombinant" in the clades output)

File: config/config.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jameshadfield) [reply config/config.yaml:131]: The code only expects a boolean `defining_lineage` for "other" at the moment when processing `pango_lineages`, so an additional one-line patch is needed. (However also see the wider-ranging discussion elsewhere in the PR.)

```diff
@@ -23,7 +23,7 @@ def order_lineages(lineages, aliasor):
 def lineage_to_clade(lineage, aliasor, fallback, clade_definitions):
     lineage_full = aliasor.uncompress(lineage)
     for clade_data in clade_definitions:
-        if clade_data['clade']=='other':
+        if clade_data['clade']=='other' or clade_data['clade']=='recombinant':
             continue
```

**Comment by joverlee521** [comment]
> Modify the upstream data categorization so that what's currently "recombinant" is grouped together with "other". This seems the most sensible to me but I am not an expert on the ingest side of things here.

Ah, this was the direction @trvrb was going in https://github.com/nextstrain/forecasts-ncov/pull/45. I can close this PR in favor of grouping "recombinant" into "other".

**Pull request closed without merging**